src/katsdpmodels/primary_beam.py

```diff
@@ -100,6 +100,13 @@ class PrimaryBeam(models.SimpleHDF5Model):
     An instance of this class describes either a single antenna or an average
     of multiple antennas; to do antenna-specific modelling you will need a
     separate instance of this class per antenna.
```

Comment:
I think you should keep your eye on perhaps a later goal (and keep the design accordingly), that one would want different beams per antenna and that these would probably be provided in the same file, along with the array average.
A bigger question is whether you want to use an array average which is 'broadened' due to (typical or measured) pointing errors or not as it depends on the application which should be used. For primary beam correction (post imaging) you need to use the broadened array average beam, while if you do on the fly A projection then you may opt to use non-broadened beams.

Reply:
I'm not keen to have multiple antennas in the same file, because it means that every time one antenna is changed, the entire multi-gigabyte file has to be replaced. I know you're working on a representation in which you'd have some data that's antenna-specific and some that's only band-specific, but that could be handled in a future file format by having the antenna files reference the band files.

Just because different antennas are in different files doesn't mean that one couldn't build an API that does joint computation across multiple files.

Broadening I think could be handled by providing two array average files, and letting the user select which to use.